### PR TITLE
SL-20442 Add an 'Until shortcut key released' option to gestures so we can do properly user-mappable keys

### PR DESCRIPTION
**Pull request by LLGuru**

### CODE REVIEW

File: indra/newview/llgesturemgr.cpp [hunk 1]
```diff
@@ -56,6 +56,8 @@
 #include "llappearancemgr.h"
 #include "llgesturelistener.h"
 
+#pragma optimize("", off)
```
Comment (akleshchev) [inline indra/newview/llgesturemgr.cpp:59]: Why is this here?

**Review by akleshchev — CHANGES_REQUESTED** [review]
Please either remove 'pragma optimize' or reduce it to scope of the code that needs it along with the explanation of why.

File: indra/newview/llgesturemgr.cpp (hunk 1 above)
Comment (marchcat-pe) [reply indra/newview/llgesturemgr.cpp:59]: We should avoid using compiler-specific code, especially in such general areas as this.

**Review by marchcat-pe — APPROVED** [review]

File: indra/newview/llgesturemgr.cpp (hunk 1 above)
Comment (LLGuru) [reply indra/newview/llgesturemgr.cpp]: Removed (there was a temporary hint for debug purposes)

**Review by akleshchev — APPROVED** [review]

**Comment by AtlasLinden** [comment]
Archived issue: https://github.com/secondlife/jira-archive-internal/issues/70809
